src/plots/cartesian/axes.js

```diff
@@ -455,6 +455,13 @@ axes.expand = function(ax, data, options) {
         i, j, v, di, dmin, dmax,
         ppadiplus, ppadiminus, includeThis, vmin, vmax;
 
+    // domain-constrained axes: base extrappad on the unconstrained
+    // domain so it's consistent as the domain changes
+    if(extrappad && ax._inputDomain) {
```

Comment:
-> tighter condition in https://github.com/plotly/plotly.js/pull/1767/commits/1271382e9ed31a238544fdc59cbe3c3f11cd9e6c